tests/unit/record_encoder/test_hybrid_record_encoder.py

```diff
@@ -0,0 +1,119 @@
+from unittest.mock import patch
+
+import pytest
+from pinecone_text.sparse import BM25Encoder
+
+from canopy.knowledge_base.models import KBDocChunk
+from canopy.knowledge_base.record_encoder import HybridRecordEncoder, DenseRecordEncoder
+from canopy.models.data_models import Query
+from tests.unit.stubs.stub_dense_encoder import StubDenseEncoder
+
+
+@pytest.fixture(scope="module")
+def documents():
+    return [KBDocChunk(id=f"doc_1_{i}",
+                       text=f"Sample document {i}",
+                       document_id=f"doc_{i}",
+                       metadata={"test": i},
+                       source="doc_1")
+            for i in range(5)]
+
+
+@pytest.fixture(scope="module")
+def queries():
+    return [Query(text="Sample query 1"),
+            Query(text="Sample query 2"),
+            Query(text="Sample query 3")]
+
+
+@pytest.fixture(scope="module")
+def inner_dimension():
+    return 4
+
+
+@pytest.fixture(scope="module")
+def dense_record_encoder(inner_dimension):
+    return DenseRecordEncoder(StubDenseEncoder(dimension=inner_dimension))
+
+
+@pytest.fixture(scope="module")
+def bm_25_encoder_df_path(documents):
+    bm25 = BM25Encoder()
+    bm25.fit([doc.text for doc in documents])
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_path = str(Path(tmp_dir, "bm25_params.json"))
+        bm25.dump(tmp_path)
+        yield tmp_path
+
+
+@pytest.fixture(scope="module")
+def hybrid_encoder(dense_record_encoder, bm_25_encoder_df_path):
+    return HybridRecordEncoder(dense_record_encoder,
+                               bm_25_encoder_df_path=bm_25_encoder_df_path,
+                               batch_size=2)
+
+
+def test_dimension(hybrid_encoder, inner_dimension):
+    assert hybrid_encoder.dimension == inner_dimension
+
+
+def test_init_encoder_invalid_alpha(dense_record_encoder):
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(dense_record_encoder, alpha=-1)
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(dense_record_encoder, alpha=2)
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(dense_record_encoder, alpha=0)
+
+
+def test_encode_documents(hybrid_encoder, documents, queries):
+    encoded_documents = hybrid_encoder.encode_documents(documents)
+    for encoded_document in encoded_documents:
+        assert len(encoded_document.values) == hybrid_encoder.dimension
+        assert "indices" in encoded_document.sparse_values
+        assert "values" in encoded_document.sparse_values
+
+
+def test_encode_queries(hybrid_encoder, queries):
+    encoded_queries = hybrid_encoder.encode_queries(queries)
+    assert len(encoded_queries) == len(queries)
+    for encoded_query in encoded_queries:
+        assert len(encoded_query.values) == hybrid_encoder.dimension
+        assert "indices" in encoded_query.sparse_values
+        assert "values" in encoded_query.sparse_values
+
+
+def test_encode_queries_alpha_applied_correctly(hybrid_encoder, queries):
```

Comment:
@izellevy I don't think this test is really testing what you're saying it tests.  

1. It's not testing that changing `alpha` as init parameter really works, since you patched over it.
2. It's not really checking that the output is `alpha * dense_value`, just that if you changed `alpha` the output also changes
3. It doesn't verify that sparse is affected by `(1 - alpha)`.

I suggest testing this feature exactly opposite: Create a new `HybridRecrordEncoder` and pass a deliberate `alpha` value to the contructor (e.g. `0.2`). Then, path the `encoder._dense_encoder.encode_queries()` and `encoder._sparse_encoder.encode_queries()` methods, controlling what they will output.  
Now you can check that calling `encoder.encode_queries()` really returns a value which is what expected.

Reply:
1) I now added created RecordEncoder with alpha
2) Right but then I kinda duplicate this test: https://github.com/pinecone-io/pinecone-text/blob/7dff87b1262f8b22df5c98ebd18528329e107b62/tests/unit/test_hybrid_convex_scale.py#L11
3) Now I also check that the scaling applied correctly to sparse.

If the sparse and dense matches with scaling I think we can assume it is calculated correctly.

Reply:
OK...